app/components/user_menu/signed_in_component.rb

```diff
@@ -24,4 +24,8 @@ def conversations?
   def admin?
     user.admin?
   end
+
+  def unread_notifications?
+    user.message_notifications&.unread&.any?
```

Comment:
Hmmm ok, I’ll read up on this so I don’t make the same error in the future. I could have sworn I was getting errors when I tried to call @current_user.message_notifications.read on an empty collection.

Reply:
It may have been an empty array, not an Active Record query result.

Reply:
Here's the error that comes up on a newly registered account (ie one for which message_notifications is totally empty) when I make that change (and why I had to add in the safe navigation in the first place): 
<img width="830" alt="Screen Shot 2021-12-31 at 12 25 46 PM" src="https://user-images.githubusercontent.com/84106309/147832364-51504043-7f81-4dec-9784-1f0ef5dad243.png">

Reply:
The reason for the error is that the user hasn't set up a developer or business profile yet, so message_notifications is nil instead of an empty query result.

Reply:
Ah! Yes, that makes sense, I forgot about that edge case. So we can do `!! user.message_notifications&.unread&.any?` which will turn a `nil` into `false.

Reply:
Actually, I take that back. We should always return an association from that method.

```
def message_notifications
  if developer.present? && business.present?
    developer_notifications.or(business_notifications).order(created_at: :desc)
  elsif developer.present?
    developer_notifications.order(created_at: :desc)
  elsif business.present?
    business_notifications.order(created_at: :desc)
  else
    Notification.none
  end
end
```

This will ensure we always get something back and can chain our scopes.

Reply:
Got it, will make those changes.